src/main/scala/org/broadinstitute/dsde/workbench/avram/util/AvramError.scala

```diff
@@ -0,0 +1,46 @@
+package org.broadinstitute.dsde.workbench.avram.util
+
+import com.google.api.server.spi.ServiceException
+import com.google.api.server.spi.response._
+import io.circe.{DecodingFailure, ParsingFailure}
+
+case class AvramError(statusCode: Int, message: String) {
+
+  def toServiceException: ServiceException = {
+    statusCode match {
+      case 400 => new BadRequestException(message)
+      case 401 => new UnauthorizedException(message)
+      case 403 => new ForbiddenException(message)
+      case 404 => new NotFoundException(message)
+      case 409 => new ConflictException(message)
+      case 500 => new InternalServerErrorException(message)
+      case 503 => new ServiceUnavailableException(message)
+      case _ =>
+        new InternalServerErrorException(message)
+    }
+  }
+
+}
+
+object AvramError {
+  // Typeclass for converting types to AvramError
+  trait AvramErrorable[A] {
```

Comment:
This lets us implicitly convert various exception types into an `AvramError`. 

Still not certain whether this is a good idea. The alternative is to require converting errors to `AvramError` at the place of use.

Maybe there is some middle ground: we could auto-convert common circe errors but not DB errors for example.